providers/gce/gce_instances.go

```diff
@@ -365,7 +365,7 @@ func (g *Cloud) InstanceMetadata(ctx context.Context, node *v1.Node) (*cloudprov
 		}
 	}
 
-	_, zone, name, err := splitProviderID(providerID)
+	project, zone, name, err := splitProviderID(providerID)
```

Comment:
Is cross project communication here a potential escalation path? Does this possibility exist elsewhere already, or is this the first place this is being added?